app/main.development.js

```diff
@@ -226,56 +264,50 @@ const installExtensions = async () => {
 };
 
 const { ipcMain } = require("electron");
-
-ipcMain.on("start-daemon", (event, appData) => {
+/*
+ipcMain.on("get-available-wallets", (event) => {
+  event.returnValue = availableWallets;
+});
+*/
+ipcMain.on("start-daemon", (event, walletPath, appData, testnet) => {
   if (dcrdPID && !daemonIsAdvanced) {
     logger.log("info", "Skipping restart of daemon as it is already running");
-    event.returnValue = dcrdPID;
     return;
   }
   if(appData){
     logger.log("info", "launching dcrd with different appdata directory");
   }
   if (dcrdPID) {
     logger.log("info", "dcrd already started " + dcrdPID);
-    event.returnValue = dcrdPID;
     return;
   }
   try {
-    dcrdPID = launchDCRD(appData);
+    dcrdPID = launchDCRD(walletPath, appData, testnet);
```

Comment:
This is what is preventing the clean shutdown. `launchDCRD` is **not** returning the pid, but a `dcrdConfig` object, and then when shutting down, the `closeDcdr()` call doesn't actually close the daemon.